Remove id, created, updated and deleted properties from BaseSortFields

### Subject of the issue
Remove id, created, updated and deleted properties from BaseSortFields and create SimpleBaseSortFields with these properties, so the developer can decide wich one to use.

### Your environment
* version of merlin_gql: 0.2.2
* version of node: 12.17.0
* version of typeorm: 0.2.25

### Steps to reproduce
Create a model with id or created or updated or deleted property, creates the SortField class with these model attributes and  extending BaseSortFiled, you get these error:

_error TS2416: Property 'deleted' in type 'TmTopicAssignmentSorts' is not assignable to the same property in base type 'BaseSortFields'._

### Expected behaviour
The BaseSortField must be a clean class to extend

### Actual behaviour
_error TS2416: Property 'deleted' in type 'TmTopicAssignmentSorts' is not assignable to the same property in base type 'BaseSortFields'._

--- a/src/core/type-graphql/models/base-filter-fields.ts
+++ b/src/core/type-graphql/models/base-filter-fields.ts
@@ -1,7 +1,5 @@
 import { Field, ID, InputType, ClassType } from "type-graphql";
 
-
-
 export class BaseFilterFields {
   
 }

--- a/src/core/type-graphql/models/base-sort-fields.ts
+++ b/src/core/type-graphql/models/base-sort-fields.ts
@@ -3,15 +3,4 @@ import { SortDirectionsEnum } from "../resolvers/query-resolver";
 
 @InputType()
 export class BaseSortFields {
-  @Field((type) => SortDirectionsEnum, { nullable: true })
-  id?: number;
-
-  @Field((type) => SortDirectionsEnum, { nullable: true })
-  created?: Date;
-
-  @Field((type) => SortDirectionsEnum, { nullable: true })
-  updated?: Date;
-
-  @Field((type) => SortDirectionsEnum, { nullable: true })
-  deleted?: boolean;
 }

--- a/src/core/type-graphql/models/simple-base-filter-fields.ts
+++ b/src/core/type-graphql/models/simple-base-filter-fields.ts
@@ -1,0 +1,16 @@
+import { Field, ID, InputType, ClassType } from "type-graphql";
+import { FilteredID, FilteredString, FilteredDate, FilteredBoolean } from '../resolvers/filter-field';
+
+export class SimpleBaseFilterFields {
+    @Field((type) => FilteredID, { nullable: true })
+    id?: number;
+  
+    @Field((type) => FilteredDate, { nullable: true })
+    created?: Date;
+  
+    @Field((type) => FilteredDate, { nullable: true })
+    updated?: Date;
+  
+    @Field((type) => FilteredBoolean, { nullable: true })
+    deleted?: boolean;
+}

--- a/src/core/type-graphql/models/simple-base-sort-fields.ts
+++ b/src/core/type-graphql/models/simple-base-sort-fields.ts
@@ -1,0 +1,17 @@
+import { Field, InputType } from "type-graphql";
+import { SortDirectionsEnum } from "../resolvers/query-resolver";
+
+@InputType()
+export class SimpleBaseSortFields {
+  @Field((type) => SortDirectionsEnum, { nullable: true })
+  id?: number;
+
+  @Field((type) => SortDirectionsEnum, { nullable: true })
+  created?: Date;
+
+  @Field((type) => SortDirectionsEnum, { nullable: true })
+  updated?: Date;
+
+  @Field((type) => SortDirectionsEnum, { nullable: true })
+  deleted?: boolean;
+}

--- a/src/core/type-graphql/resolvers/filter-field.ts
+++ b/src/core/type-graphql/resolvers/filter-field.ts
@@ -1,7 +1,25 @@
 import { GraphQLScalarType } from "graphql";
-import { ClassType, InputType, Field, ID } from "type-graphql";
+import { ClassType, InputType, Field, ID, Int, Float, GraphQLISODateTime } from "type-graphql";
 import { FilterTypesEnum } from "./query-resolver";
 
+
+@InputType()
+export class FilteredID extends FilterField(ID) {}
+
+@InputType()
+export class FilteredString extends FilterField(String) {}
+
+@InputType()
+export class FilteredInt extends FilterField(Int) {}
+
+@InputType()
+export class FilteredFloat extends FilterField(Float) {}
+
+@InputType()
+export class FilteredBoolean extends FilterField(Boolean) {}
+
+@InputType()
+export class FilteredDate extends FilterField(GraphQLISODateTime) {}
 
 export default function FilterField<TField>(
   TFieldClass: ClassType<TField> | GraphQLScalarType

